simulation/traffic_simulator/test/src/behavior/test_longitudinal_speed_planner.cpp

```diff
@@ -0,0 +1,303 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <cmath>
+#include <traffic_simulator/behavior/longitudinal_speed_planning.hpp>
+
+#include "../expect_eq_macros.hpp"
+
+#define EXPECT_CONSTRAINTS_BOUNDED(DATA, lower, upper) \
+  EXPECT_GE(DATA.max_speed, lower);                    \
+  EXPECT_GE(DATA.max_acceleration, lower);             \
+  EXPECT_GE(DATA.max_deceleration, lower);             \
+  EXPECT_GE(DATA.max_acceleration_rate, lower);        \
+  EXPECT_GE(DATA.max_deceleration_rate, lower);        \
+  EXPECT_LE(DATA.max_speed, upper);                    \
+  EXPECT_LE(DATA.max_acceleration, upper);             \
+  EXPECT_LE(DATA.max_deceleration, upper);             \
+  EXPECT_LE(DATA.max_acceleration_rate, upper);        \
+  EXPECT_LE(DATA.max_deceleration_rate, upper);
+
+int main(int argc, char ** argv)
+{
+  testing::InitGoogleTest(&argc, argv);
+  return RUN_ALL_TESTS();
+}
+
+auto makeLongitudinalSpeedPlanner()
+  -> traffic_simulator::longitudinal_speed_planning::LongitudinalSpeedPlanner
+{
+  const double step_time = 0.001;
+  const std::string entity = "entity";
+  return traffic_simulator::longitudinal_speed_planning::LongitudinalSpeedPlanner{
+    step_time, entity};
+}
+
+TEST(LongitudinalSpeedPlanner, isAccelerating_true)
+{
+  auto planner = makeLongitudinalSpeedPlanner();
+
+  geometry_msgs::msg::Twist current_twist{};
+  current_twist.linear.x = 1.0;
+
+  const double target_speed = 2.0;
+  EXPECT_TRUE(planner.isAccelerating(target_speed, current_twist));
+}
+
+TEST(LongitudinalSpeedPlanner, isAccelerating_false)
+{
+  auto planner = makeLongitudinalSpeedPlanner();
+
+  geometry_msgs::msg::Twist current_twist{};
+  current_twist.linear.x = 3.0;
+
+  const double target_speed = 2.0;
+  EXPECT_FALSE(planner.isAccelerating(target_speed, current_twist));
+}
+
+TEST(LongitudinalSpeedPlanner, isDecelerating_true)
+{
+  auto planner = makeLongitudinalSpeedPlanner();
+
+  geometry_msgs::msg::Twist current_twist{};
+  current_twist.linear.x = 3.0;
+
+  const double target_speed = 2.0;
+  EXPECT_TRUE(planner.isDecelerating(target_speed, current_twist));
+}
+
+TEST(LongitudinalSpeedPlanner, isDecelerating_false)
+{
+  auto planner = makeLongitudinalSpeedPlanner();
+  geometry_msgs::msg::Twist current_twist{};
+  current_twist.linear.x = 1.0;
+
+  const double target_speed = 2.0;
+  EXPECT_FALSE(planner.isDecelerating(target_speed, current_twist));
+}
+
+TEST(LongitudinalSpeedPlanner, getAccelerationDuration_acceleration)
+{
+  auto planner = makeLongitudinalSpeedPlanner();
+
+  geometry_msgs::msg::Twist current_twist{};
+  geometry_msgs::msg::Accel current_accel{};
+  traffic_simulator_msgs::msg::DynamicConstraints constraints{};
+  current_twist.linear.x = 1.0;
+  current_accel.linear.x = 1.0;
+  constraints.max_speed = 10.0;
+  constraints.max_acceleration = 2.0;
+  constraints.max_acceleration_rate = 1.0;
```

Comment:
Please consider using `geometry_msgs::msg::build<message_type>` function for building message.
Avoid creating unnecessary temporary variables for readability.